system/jlib/jlog.hpp

```diff
@@ -1335,4 +1335,183 @@ interface IComponentLogFileCreator : extends IInterface
 extern jlib_decl IComponentLogFileCreator * createComponentLogFileCreator(IPropertyTree * _properties, const char *_component);
 extern jlib_decl IComponentLogFileCreator * createComponentLogFileCreator(const char *_logDir, const char *_component);
 extern jlib_decl IComponentLogFileCreator * createComponentLogFileCreator(const char *_component);
+
+enum AccessLogRangeType
+{
+    ACRTUnknown = -1,
+    ACRTAllAvailable,
+    ACRTFirstPage,
+    ACRTirstNRows,
+    ACRTLastNYears,
+    ACRTLastNMonths,
+    ACRTLastNWeeks,
+    ACRTLastNDays,
+    ACRTLastNHours,
+    ACRTLastNMinutes,
+    ACRTLastPage,
+    ACRTGoToPage,
+    ACRTLastNRows,
+    ACRTTimeRange,
+};
+
+struct LogAccessRange
+{
+    AccessLogRangeType rangeType;
+    StringBuffer startDate;
+    StringBuffer endDate;
+    unsigned quantity;
+};
+
+enum AccessLogFilterType
+{
+    ALF_ALLAvailable = 0,
+    ALF_Component = 1,
+    ALF_WorkUnit = 2,
+    ALF_Audience = 3,
+    ALF_Class = 4,
+    ALF_NativeQuery = 5
+};
+
+struct LogAccessFilter
+{
+    AccessLogFilterType filterType;
+    StringBuffer workUnit;
+    StringArray targetAudience;
+    StringArray messageClass;
+    StringBuffer componentName;
+};
+
+struct LogAccessConditions
+{
+    LogAccessFilter filter;
```

Comment:
I originally envisioned supporting a tree structure to support complex filter structures, but then struggled to find use cases that justified the added complexity and chose to support basic filters (jobid="xyz", or class="error") along with a time-frame because it seemed to cover most situations.
Before I add the logic to track filter via trees, I'd like to get your opinion on how often you envision complex filter being used. @jakesmith

Reply:
Provided mechanism to support compound filters. ES plugin doesn't currently support it.